.github/workflows/macos_tests.yml

```diff
@@ -0,0 +1,82 @@
+name: MacOS tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+  release:
+  workflow_dispatch:
+
+jobs:
+  tests:
+    name: Test - ${{ matrix.os }} - Build type ${{ matrix.build_type }} - Build testing ${{ matrix.build_testing }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-11
+          - macos-12
+          # TODO: can change to macos-latest (but it's currently aliased to macos-11)
+        build_type:
+          - Release
+        build_testing:
+          - ON
+          - OFF
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Setup cache
+        id: cache-test-data
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/tdms/tests/system/data
+          key: test_data
+
+      - name: Set up MATLAB
+        uses: matlab-actions/setup-matlab@v1
+
+      - name: Install dependencies
+        run: |
+          brew install fftw llvm libomp python3
+
+      - name: Install Python dependencies
+        shell: bash
+        run: |
+          pip3 install -r ${GITHUB_WORKSPACE}/tdms/tests/requirements.txt
+
+      - name: Create Build Environment
+        run: |
+          cmake -E make_directory ${{runner.workspace}}/build
+
+      - name: Configure TDMS
+        shell: bash
+        working-directory: ${{runner.workspace}}/build
+        env:
+          CXXFLAGS: -Werror
+        run: |
+          cmake ${GITHUB_WORKSPACE}/tdms -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON\
+           -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}\
+           -DBUILD_TESTING=${{ matrix.build_testing }}\
+           -DOMP_ROOT=/usr/local/opt/llvm/ -DCXX_ROOT=/usr/local/opt/llvm
+
+      - name: Build TDMS
+        working-directory: ${{runner.workspace}}/build
+        shell: bash
+        run: |
+          cmake --build . --config ${{ matrix.build_type }}
+
+      - name: Run TDMS unit tests
+        if: matrix.build_testing == 'ON'
+        working-directory: ${{runner.workspace}}/build
+        run: |
+          export OMP_NUM_THREADS=1
+          ctest -C ${{ matrix.build_type }} --extra-verbos
+
+      - name: Run TDMS system tests
+        working-directory: ${{runner.workspace}}/build
```

Comment:
```suggestion
      - name: Run TDMS system tests
        if: ${{ github.ref == 'refs/heads/main' }}
        # run system tests in mac only on pushes to master (they are a bit slow)
        working-directory: ${{runner.workspace}}/build
```

Reply:
OK this works. Is the strategy with you, @giordano ?

Reply:
I agree the much longer wall time isn't great, but do we have an intuition of why tests on macOS are much slower than the rest?

Also, to reduce the usage of resources, can we perhaps test only on `macos-latest`?  Although not impossible, we'd need to be particularly unlucky to hit some version-specific peculiarities for this kind of software.

Reply:
I've had a small amount of experience with running MacOS jobs on gh runners and it seems like the hardware is just much slower

> Also, to reduce the usage of resources, can we perhaps test only on macos-latest? 

+1 on this

Reply:
`macos-latest` is `macos-11`. I would actually rather test 12. Hence my leaving the TODO.
Let me remove 11.

Reply:
> but do we have an intuition of why tests on macOS are much slower than the rest?

Seems to be an actual issue: actions/runner-images#3885